### PR TITLE
Database health check deactivation via configuration

### PR DESCRIPTION
**Pull request by NikolaDmitrasinovic**
# Readiness checklist

-   [ ] I added/updated unit tests.
-   [ ] I added/updated integration tests.
-   [x] I ensured that the PR title is good enough for the changelog.
-   [x] I labeled the PR.

### CODE REVIEW

File: ConsumerApi/Extensions/IServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -100,7 +100,7 @@ public static IServiceCollection AddCustomAspNetCore(this IServiceCollection ser
 
         services.AddTransient<IUserContext, AspNetCoreUserContext>();
 
-        if (configuration.Modules.Devices.Infrastructure.PushNotifications.Providers.Sse is { Enabled: true })
+        if (configuration.Modules.Devices.Infrastructure.PushNotifications.Providers.Sse is { Enabled: true, EnableHealthCheck: true })
```
Comment (tnotheis) [inline ConsumerApi/Extensions/IServiceCollectionExtensions.cs:103]: It shouldn't be possible to disable the SSE Server health check

File: ConsumerApi/appsettings.json [hunk 2]
```diff
@@ -14,16 +14,30 @@
                 "MinimumBackoff": 2,
                 "MaximumBackoff": 120
             }
+        },
+        "SqlDatabaseConfiguration": {
```
Comment (tnotheis) [inline ConsumerApi/appsettings.json:18]: There is no `SqlDatabaseConfiguration` property

```suggestion
        "SqlDatabase": {
```

File: ConsumerApi/appsettings.json [hunk 3]
```diff
@@ -14,16 +14,30 @@
...
+            "EnableHealthCheck": false
```
Comment (tnotheis) [inline ConsumerApi/appsettings.json:19]: All db health checks should be enabled by default

File: Modules/Challenges/src/Challenges.ConsumerApi/ChallengesModule.cs [hunk 4]
```diff
@@ -29,7 +29,10 @@ public override void ConfigureServices(IServiceCollection services, IConfigurati
             dbOptions.DbConnectionString = parsedConfiguration.Infrastructure.SqlDatabase.ConnectionString;
         });
 
-        services.AddSqlDatabaseHealthCheck(Name, parsedConfiguration.Infrastructure.SqlDatabase.Provider, parsedConfiguration.Infrastructure.SqlDatabase.ConnectionString);
+        services.AddSqlDatabaseHealthCheck(Name,
+            parsedConfiguration.Infrastructure.SqlDatabase.Provider,
+            parsedConfiguration.Infrastructure.SqlDatabase.ConnectionString,
+            parsedConfiguration.Infrastructure.SqlDatabase.EnableHealthCheck);
```
Comment (tnotheis) [inline Modules/Challenges/src/Challenges.ConsumerApi/ChallengesModule.cs:35]: I don't like that we call `AddSqlDatabaseHealthCheck`, pass `EnableHealthCheck`, and let the method decide whether it ACTUALLY enables add the sql database health check or not. That's not intuitive at all.
Instead, we should probably perform the check outside of the method. 
Same for all occurrences, of course.

File: Modules/Devices/src/Devices.Infrastructure/PushNotifications/Connectors/Sse/SseOptions.cs [hunk 5]
```diff
@@ -7,6 +7,9 @@ public class SseOptions
     [Required]
     public required bool Enabled { get; set; }
 
+    [Required]
+    public bool EnableHealthCheck { get; set; } = true;
```
Comment (tnotheis) [inline Modules/Devices/src/Devices.Infrastructure/PushNotifications/Connectors/Sse/SseOptions.cs:11]: Remove this (it shouldn't be possible to disable the sse server health check)

**Review by tnotheis — CHANGES_REQUESTED** [review]

File: Modules/Devices/src/Devices.Infrastructure/PushNotifications/Connectors/Sse/SseOptions.cs (hunk 5 above)
Comment (NikolaDmitrasinovic) [reply Modules/Devices/src/Devices.Infrastructure/PushNotifications/Connectors/Sse/SseOptions.cs:11]: removed

File: ConsumerApi/appsettings.json (hunk 3 above)
Comment (NikolaDmitrasinovic) [reply ConsumerApi/appsettings.json:19]: done

File: ConsumerApi/Extensions/IServiceCollectionExtensions.cs (hunk 1 above)
Comment (NikolaDmitrasinovic) [reply ConsumerApi/Extensions/IServiceCollectionExtensions.cs:103]: done

File: Modules/Challenges/src/Challenges.ConsumerApi/ChallengesModule.cs (hunk 4 above)
Comment (NikolaDmitrasinovic) [reply Modules/Challenges/src/Challenges.ConsumerApi/ChallengesModule.cs]: done

**Review by tnotheis — APPROVED** [review]

**Pull request merged**